packages/block-editor/src/store/reducer.js

```diff
@@ -1121,8 +1121,8 @@ export function blocksMode( state = {}, action ) {
 export function insertionPoint( state = null, action ) {
 	switch ( action.type ) {
 		case 'SHOW_INSERTION_POINT':
-			const { rootClientId, index } = action;
-			return { rootClientId, index };
+			const { rootClientId, index, canReplaceBlock } = action;
```

Comment:
Hi @ceyhun, it seems canReplaceBlock is not used as part of the state logic in the reducer, it just passed in action so it can be retrieved from the state later.
I wonder if instead of storing canReplaceBlock in the state it would be possible to call something equivalent to getShouldReplaceBlock() when we need this information?

Reply:
This goes in a similar direction to what @Tug was recommending.

Reply:
Hi @jorgefilipecosta, it is a state deriving from user picking “Add Before/After“ or “Replace” and it is kept locally in the `Inserter`. Later on it’s used in `BlockList` to hide the block that could be replaced. I’m thinking about another approach. Please see my other comment [here](https://github.com/WordPress/gutenberg/pull/18791#discussion_r358707205).